src/evo/providertx.cpp

```diff
@@ -254,6 +258,31 @@ bool CheckUpdateAssetTx(const CTransaction &tx, const CBlockIndex *pindexPrev, C
         return state.DoS(100, false, REJECT_INVALID, "bad-assets-collateralAddress");
     }
 
+    if ((assetTx.referenceHash.length() > 128)) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-assets-referenceHash");
+    }
+
+    if (!validateAmount(assetTx.amount, asset.decimalPoint)) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-assets-amount");
+    }
+
+    if (assetTx.issueFrequency <= 0 && assetTx.type != 0) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-assets-issueFrequency");
+    }
+
+    if (assetTx.maxMintCount < asset.mintCount) {
+        return state.DoS(100, false, REJECT_INVALID, "bad-assets-maxMintCount");
+    }
+    //when transferring ownership don't allow any other change
+    if (assetTx.ownerAddress != asset.ownerAddress) {
```

Comment:
if this is ownership transfer, should ownerAddress be different?

Reply:
yes, as ownership is defined by the ownerAddress, they need to be different if we are transferring the ownership of the asset